src/graphql/resolvers/box-resolver.ts

```diff
@@ -34,6 +35,19 @@ class BoxesQueryArgs {
   @Field(() => String, { nullable: true })
   boxId?: string;
 
+  @ValidateIf((o: BoxesQueryArgs) => {
+    const indexFields = [
+      o.boxId,
+      o.transactionId,
+      o.headerId,
+      o.address,
+      o.ergoTree,
+      o.ergoTreeTemplateHash
+    ];
+    const definedCount = indexFields.filter((el) => isDefined(el)).length;
+    return !(o.spent === true && definedCount > 0);
+  })
+  @IsEmpty({ message: "Registers filter should be used with spent:true and at least one index." })
```

Comment:
I think that is better to individually name index fields, just using "index" can be a bit confusing for the users.